File: 2024_07_July/30_Thursday/container_with_most_water.go
func maxArea(height []int) int {
    max_area, left, right := 0, 0, len(height) - 1

    for left < right {
        max_area = max(max_area, (right - left) * min(height[left], height[right]))

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }

    return max_area
}

File: 2024_07_July/30_Thursday/container_with_most_water.cpp
class Solution
{
public:
  int maxArea(vector<int> &height)
  {
    int max_area = 0;
    int left = 0;
    int right = height.size() - 1;

    while (left < right)
    {
      max_area = max(max_area, (right - left) * min(height[right], height[left]));

      if (height[left] < height[right])
      {
        left++;
      }
      else
      {
        right--;
      }
    }

    return max_area;
  }
};